### PR TITLE
Initial version of Extended Support Library.

### PR DESCRIPTION
**Pull request by SidLeung**
The Extended Support Library is a collection of libraries that support features that are additional to the Arduino 101 platform.

Additions:
1.  Quark I2C interface library.  Support I2C0 and I2C1 of the Quark Core.

### CODE REVIEW

**Comment by bbaltz505** [comment]
I think there are some backup files checked in by mistake (*~)

**Comment by SidLeung** [comment]
@bigdinotech @eriknyquist Would you please review this library to support the Quark I2C interface.  Thanks.

File: examples/master_check/master_check.ino [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+static int count = 0;
+static uint8_t buffer_sender[MY_BUFFER_SIZE]; // data source for master writer
+static uint8_t buffer_receiver[MY_BUFFER_SIZE]; // data distination for master reader
+
+void setup() 
+{
+  I2C0.begin();         // join i2c bus 
+  Serial.begin(115200); // start serial for output
+  while (Serial)
```
Comment (bigdinotech) [inline examples/master_check/master_check.ino:33]: i think you meant:
`while(!Serial);`

File: examples/master_reader/master_reader.ino [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ *    4. The sketch is accessing an external I2C device with an device number (address) of 8.  This address can be
+ *       changed to fit your setup.
+ */
+
+#include <quarkI2C.h>
+
+void setup()
+{
+  Serial.begin(115200);  // start serial for output
+  while(Serial);
```
Comment (bigdinotech) [inline examples/master_reader/master_reader.ino:25]: same thing here

File: examples/master_writer/master_writer.ino [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ *       changed to fit your setup.
+ */
+
+
+#include <quarkI2C.h>
+
+void setup()
+{
+  Serial.begin(115200);
+  while(Serial);
```
Comment (bigdinotech) [inline examples/master_writer/master_writer.ino:26]: same thing here

**Pull request closed without merging**